pkg/api/responses.go

```diff
@@ -69,3 +72,16 @@ var responseOperationInvalid = []byte(
 	`{ "error": "OperationInvalid", "description": "The polling request ` +
 		`included an invalid value for the required operation query parameter" }`,
 )
+
+var responseTagsMalformed = []byte(`{ "error": "MalformedRequestTags", ` +
+	`"description": The provided tags were not well-formed JSON" }`,
+)
+
+var responseIncorrectRequestBody = []byte(`{ "error": "MalformedRequest", ` +
+	`"description": The provided request did not match what was expected for ` +
+	`the service" }`,
+)
+
+var responseValidationFailedTemplate = string(`{ "error": "ValidationError", ` +
+	`"description": The value provided for %s is invalid. %s" }`,
+)
```

Comment:
Could we replace this with a function that does the formatting so users of `responseValidationFailedTemplate` aren't all individually responsible for that? Maybe make it that we pass it a validation error and it returns the response body?

If we were to do that, should we use functions to access the _other_ response bodies as well? I'm thinking yes, both on account of consistency and the likelihood that we might want to further parameterize all of these responses.

Reply:
Some kind of escaping might also be needed on the invalid value. What if it contains a double quote? It would mess up the well-formedness (is that a word?) of the response.

Could it possibly be better to use a struct (even an anonymous one) to encapsulate the details of this response and then we just marshal it to JSON and count on the marshaler to get all the escaping right?

Reply:
👍 * 2